models/classes/search/tasks/UpdateResourceInIndex.php

```diff
@@ -64,7 +65,11 @@ public function __invoke($params): Report
         /** @var Search $searchService */
         $searchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
 
-        $numberOfIndexed = $searchService->index([$indexDocument]);
+        try {
+            $numberOfIndexed = $searchService->index([$indexDocument]);
+        } catch (Exception $exception) {
+            return new Report(Report::TYPE_ERROR, $exception->getMessage());
```

Comment:
You can use static method:
```suggestion
            return Report::createFailure($exception->getMessage());
```